ui/StatusQ/src/StatusQ/Controls/StatusImageSelector.qml

```diff
@@ -0,0 +1,162 @@
+        id: d
+
+        readonly property int buttonsInsideOffset: 5
+        readonly property int imageSelectorPadding: 75
+    }
+
+    contentItem: ColumnLayout {
+        id: rootImageSelector
+
+        spacing: 16
+
+        StatusBaseText {
+            id: label
+
+            elide: Text.ElideRight
+            font.pixelSize: Theme.primaryTextFontSize
+        }
+
+        Rectangle {
+            id: imageSelector
+
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            radius: 8
+            color: Theme.palette.baseColor2
+
+            StatusRoundButton {
+                id: addButton
+
+                z: parent.z + 1
+                icon.name: "add"
+                x: imageSelector.width - 2 * addButton.width / 3 - d.buttonsInsideOffset
+                y: - addButton.height / 3 + d.buttonsInsideOffset
+                type: StatusRoundButton.Type.Secondary
+                onClicked: fileDialog.open()
+            }
+
+            ColumnLayout {
+                id: uploadTextPanel
+
+                width: parent.width
+                anchors.centerIn: parent
+
+                StatusIcon {
+                    icon: "images_icon"
+                    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                    Layout.preferredWidth: 20
+                    Layout.preferredHeight: 18
+                    color: Theme.palette.baseColor1
+                    fillMode: Image.PreserveAspectFit
+                }
+
+                StatusBaseText {
+                    id: uploadText
+
+                    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                    Layout.topMargin: 5
+                    Layout.preferredWidth: parent.width - 2 * d.imageSelectorPadding
+                    font.pixelSize: Theme.primaryTextFontSize
+                    color: Theme.palette.baseColor1
+                    wrapMode: Text.WordWrap
+                    lineHeight: 1.2
+                    horizontalAlignment: Text.AlignHCenter
+                }
+
+                StatusBaseText {
+                    id: additionalText
+
+                    visible: !!root.additionalText
+                    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                    Layout.preferredWidth: parent.width - 2 * d.imageSelectorPadding
+                    font.pixelSize: Theme.secondaryTextFontSize
+                    color: Theme.palette.baseColor1
+                    wrapMode: Text.WordWrap
+                    lineHeight: 1.2
+                    horizontalAlignment: Text.AlignHCenter
+                }
+            }
+
+            Image {
+                id: image
+
+                anchors.fill: parent
+                fillMode: Image.PreserveAspectFit
+            }
+        }
+    }
+
+    FileDialog {
+        id: fileDialog
+
+        folder: shortcuts.pictures
+        nameFilters: [ qsTr("Supported image formats (%1)").arg("*.jpg *.jpeg *.jfif *.webp *.png *.heif") ]
```

Comment:
We have a list of extensions in Constants:
```suggestion
        nameFilters: [
            qsTr("Image files (%1)").arg(Constants.acceptedDragNDropImageExtensions.map(img => "*" + img).join(" "))
        ]
```

Reply:
Also, why not SVG too? :)

Reply:
Since it is a `statusq` component I've added a property where the consumer can decide the collection of file's extension.
As suggested here, in this case we use `acceptedImageExtensions: Constants.acceptedDragNDropImageExtensions`.